johnnyfive/slack.py

```diff
@@ -0,0 +1,179 @@
+                # You could also use a blocks[] array to send richer content
+            )
+            # Print result, which includes information about the message (like TS)
+            #print(result)
+        except SlackApiError as e:
+            print(f"Error: {e}")
+        return response
+
+    def upload_file(self, file, title=None):
+        """upload_file Upload a file to the channel
+
+        _extended_summary_
+
+        Parameters
+        ----------
+        file : `str` or `os.PathLike`
+            The (path and) filename of the file to be uploaded.
+        title : `str`, optional
+            The title for the file posted [Default: None]
+
+        Returns
+        -------
+        `Any`
+            The response from Slack
+        """
+        response = None
+        try:
+            response = self.client.files_upload(
+                                    channels=self.channel_id,
+                                    file=file,
+                                    title=title)
+        except SlackApiError as e:
+            print(f"Error: {e}")
+        return response
+
+    def _read_channels(self, name):
+        """_read_channels Return the Channel ID for the names channel
+
+        Parameters
+        ----------
+        name : `str`
+            The name of the channel
+
+        Returns
+        -------
+        `str`
+            The desired Channel ID
+        """
+        conversation_id = None
+
+        try:
+            # Call the conversations.list() method using the WebClient
+            for _ in (result := self.client.conversations_list()):
+                if conversation_id is not None:
+                    break
+                for channel in result["channels"]:
+                    if channel["name"] == name:
+                        conversation_id = channel["id"]
+                        #Print result
+                        print(f"Found conversation ID: {conversation_id}")
+                        break
+
+        except SlackApiError as e:
+            print(f"Error: {e}")
+
+        # Return the conversation ID
+        return conversation_id
+
+
+# Internal Functions =========================================================#
+def setup_slack():
+    """setup_slack Setup the Slack WebClient for posting
+
+    _extended_summary_
+
+    Returns
+    -------
+    client : `slack_sdk.web.client.WebClient`
+        The WebClient object needed for reading and writing
+    logger : `logging.Logger`
+        The logging thingie
+    """
+    # Read the setup
+    setup = utils.read_ligmos_conffiles('slackSetup')
+
+    # SlackWebClient instantiates a client that can call API methods
+    # When using Bolt, you can use either `app.client` or the `client` passed to listeners.
+    client = SlackWebClient(token=setup.password)
+    logger = logging.getLogger(__name__)
+
+    return client, logger
+
+
+# Main Testing Driver ========================================================#
+def main():
```

Comment:
Pull out and put in an examples/ directory before merging

Reply:
Done.